dot/types/inherents.go

```diff
@@ -64,6 +68,29 @@ func (d *InherentsData) SetInt64Inherent(key []byte, data uint64) error {
 	return nil
 }
 
+// SetStructInherent sets a struct inherent.
+func (d *InherentsData) SetStructInherent(key []byte, value interface{}) error {
+	if len(key) != 8 {
+		return errors.New("inherent key must be 8 bytes")
+	}
+
+	data, err := scale.Marshal(value)
+	if err != nil {
+		return err
+	}
+
+	venc, err := scale.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	kb := [8]byte{}
+	copy(kb[:], key)
+
+	d.data[kb] = venc
```

Comment:
I don't think we should be storing the scale encoded version of this map in `InherentsData`.  I agree we should be supporting maps in the scale package. In the meantime move this code to `InherentsData.Encode()` which already loops through the map and is already called whenever we need to scale encode `InherentsData`.